client/clientBean.go

```diff
@@ -43,7 +43,7 @@ type (
 	Bean interface {
 		GetHistoryClient() history.Client
 		GetHistoryPeers() history.PeerResolver
-		GetMatchingClient(domainIDToName DomainIDToNameFunc) (matching.Client, error)
+		GetMatchingClient(domainIDToName DomainIDToNameFunc, shardDistributorClient sharddistributor.Client) (matching.Client, error)
```

Comment:
see the other comment. this interface change shouldn't be needed when `sharddistributor.Client` is an implementation detail of existing peer resolver.